File: Python/zh-cn/Lesson01/AlienFinder.py
from boa.builtins import concat
from boa.interop.Neo.Header import GetConsensusData
from boa.interop.Neo.Runtime import Notify
from boa.interop.Neo.Storage import Get, Put, GetContext
from boa.interop.Neo.Blockchain import GetHeader, GetHeight

def RandomNumber(blockHeight):
    return GetConsensusData(GetHeader(blockHeight))


def FindXna(randomNumber):
    return randomNumber % 100000000


def generateAlien(xna, alienName, blockHeight):

    context = GetContext()
    counter = Get(context, 'counter')
    if not counter:
        Put(context, 'counter', 1)
    else:
        Put(context, 'counter', counter + 1)

    temp_alien = {
        'xna': xna,
        'alienName': alienName,
        'blockHeight': blockHeight,
        'id': counter
    }

    context = GetContext()
    Put(context, counter, temp_alien)
    counter += 1
    return temp_alien


def Main(temp_alienName):
    temp_blockHeight = GetHeight()
    temp_xna = FindXna(RandomNumber(temp_blockHeight))
    someAlien = generateAlien(temp_xna, temp_alienName, temp_blockHeight)
    Notify(concat("Alien created, ID: ", someAlien['id']))

File: C#/zh-cn/Lesson01/AlienFinder.cs
using System;
using System.Numerics;
using Neo.Smartcontract.Framework;
using Neo.SmartContract.Framework.Services.Neo;

public class AlienFinder : Smartcontract {
    public class Alien {
        public static uint counter = 0;
        public uint xna;
        public string alienName;
        public uint blockHeight;
        public uint id;

        public Alien (uint xna, string alienName, uint blockHeight) {
            this.xna = xna;
            this.alienName = alienName;
            this.blockHeight = blockHeight;
            counter = counter + 1;
            this.id = counter;
        }
    }

    public static void Main (string alienName) {
        uint blockHeight = Blockchain.GetHeight ();
        uint xna = FindXna (RandomNumber (blockHeight));
        Alien someAlien = new Alien (xna, alienName, blockHeight);
        Runtime.Notify ("Alien created, ID: " + someAlien.id);
    }

    private static ulong RandomNumber (uint blockHeight) {
        return Blockchain.GetHeader (blockHeight).ConsensusData;
    }

    private static uint FindXna (ulong randomNumber) {
        return (uint) (randomNumber % 100000000);
    }

}